cmd/cstorvolume-claim/cvc/cstorvolumeclaim.go

```diff
@@ -0,0 +1,399 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cvc
+
+import (
+	"strconv"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	csp "github.com/openebs/maya/pkg/cstor/pool/v1alpha3"
+	cv "github.com/openebs/maya/pkg/cstor/volume/v1alpha1"
+	cvr "github.com/openebs/maya/pkg/cstor/volumereplica/v1alpha1"
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	svc "github.com/openebs/maya/pkg/kubernetes/service/v1alpha1"
+	sc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var (
+	ports = []corev1.ServicePort{
+		corev1.ServicePort{
+			Name:     "cstor-iscsi",
+			Port:     3260,
+			Protocol: "TCP",
+			TargetPort: intstr.IntOrString{
+				IntVal: 3260,
+			},
+		},
+		corev1.ServicePort{
+			Name:     "cstor-grpc",
+			Port:     7777,
+			Protocol: "TCP",
+			TargetPort: intstr.IntOrString{
+				IntVal: 7777,
+			},
+		},
+		corev1.ServicePort{
+			Name:     "mgmt",
+			Port:     6060,
+			Protocol: "TCP",
+			TargetPort: intstr.IntOrString{
+				IntVal: 6060,
+			},
+		},
+		corev1.ServicePort{
+			Name:     "exporter",
+			Port:     9500,
+			Protocol: "TCP",
+			TargetPort: intstr.IntOrString{
+				IntVal: 9500,
+			},
+		},
+	}
+)
+
+// getStorageClass return storageclass object for a given storageClass Name.
+// or error if any.
+func getStorageClass(storageClassName string,
+) (*storagev1.StorageClass, error) {
+	if storageClassName == "" {
+		return nil, errors.New("failed to get storageclass: storageclass missing")
+	}
+	scObj, err := sc.NewKubeClient().Get(storageClassName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"failed to get storageclass {%s}",
+			storageClassName,
+		)
+	}
+	return scObj, nil
+}
+
+// getReplicationFactor gets the Replication and Consistency factor from the
+// storageclass
+func getReplicationFactor(scName string,
+) (int, int, error) {
+
+	scObj, err := getStorageClass(scName)
+	if err != nil {
+		return 0, 0, errors.Wrapf(
+			err,
+			"failed to get storageclass obj {%s}",
+			scName,
+		)
+	}
+	count := scObj.Parameters["replicaCount"]
+
+	rfactor, err := strconv.Atoi(count)
+	if err != nil {
+		return 0, 0, errors.Wrapf(
+			err,
+			"failed to convert to int {%s}",
+			count,
+		)
+	}
+	return rfactor, (rfactor/2 + 1), nil
+
+}
+
+// getFromParameters gets replicaCount and storagePoolClaim from
+// storageclass parameter
+func getFromParameters(scName string,
+) (int, string, error) {
+
+	scObj, err := getStorageClass(scName)
```

Comment:
instead of doing 'get' for storage class multiple times, can this be retrieved in starting and pass its object around the functions?